lib/hexpm/repository/packages.ex

```diff
@@ -92,6 +92,30 @@ defmodule Hexpm.Repository.Packages do
     |> Enum.into(%{})
   end
 
+  @doc """
+    Retrieves package downloads including all views available.
+    i.e day, week, month, ninety_days, all.
+    The return value is a map with the following structure:
+    %{ package.id => %{ view1 => numDownloads, view2 => numDownloads, ...}
+    Example: %{ 105 => %{ "all" => 1_234_567, "ninety_days" => 10_000, etc... }
+  """
+  def packages_downloads_with_all_views(packages) do
+    PackageDownload.packages_and_all_download_views(packages)
+    |> Repo.all()
+    |> Enum.into([])
+    |> Enum.reduce(%{}, fn(package, acc) ->
+      {id, view, dls} = package
+      case Map.has_key?(acc, id) do
```

Comment:
Using `Map.update/4` should simplify this.

Reply:
Thanks!  Using Map.update/4 reduced this peace into two or three lines.